sqlalchemy_bigquery/_helpers.py

```diff
@@ -24,19 +25,46 @@
 )
 
 
-def google_client_info():
-    user_agent = USER_AGENT_TEMPLATE.format(sqlalchemy.__version__)
+def google_client_info(user_agent=None):
+    """
+    Return a client_info object, with an optional user agent
+    string.  If user_agent is None, use a default value.
+    """
+
+    if user_agent is None:
+        user_agent = USER_AGENT_TEMPLATE.format(sqlalchemy.__version__)
     return client_info.ClientInfo(user_agent=user_agent)
 
 
 def create_bigquery_client(
-    credentials_info=None,
-    credentials_path=None,
-    credentials_base64=None,
-    default_query_job_config=None,
-    location=None,
-    project_id=None,
-):
+    credentials_info: Optional[dict] = None,
+    credentials_path: Optional[str] = None,
+    credentials_base64: Optional[str] = None,
+    default_query_job_config: Optional[google.cloud.bigquery.job.QueryJobConfig] = None,
+    location: Optional[str] = None,
+    project_id: Optional[str] = None,
+    user_agent: Optional[str] = None,
+) -> google.cloud.bigquery.Client:
+    """Construct a BigQuery client object.
+
+    Args:
+        credentials_info Optional[dict]:
+        credentials_path Optional[str]:
+        credentials_base64 Optional[str]:
+        default_query_job_config (Optional[google.cloud.bigquery.job.QueryJobConfig]):
+            Default ``QueryJobConfig``.
+            Will be merged into job configs passed into the ``query`` method.
+        location (Optional[str]):
+            Default location for jobs / datasets / tables.
+        project_id (Optional[str]):
+            Project ID for the project which the client acts on behalf of.
+        user_agent (Optional[google.api_core.client_info.ClientInfo]):
```

Comment:
NIT:

Looks like it should be string?

```suggestion
        user_agent (Optional[str]):
```

Reply:
This line is correct. The other line that had `str` was in error.

`google_client_info` returns a `ClientInfo` object which may include the `user_agent` string as well as several other attributes:

```
self.python_version = python_version
self.grpc_version = grpc_version
self.api_core_version = api_core_version
self.gapic_version = gapic_version
self.client_library_version = client_library_version
self.user_agent = user_agent
self.rest_version = rest_version
```        

In addition, looking at the `bigquery.Client` object, it is expecting a ClientInfo object:

```
client_info (Optional[google.api_core.client_info.ClientInfo]):
```

Reply:
Just looking at the code, `user_agent` is passed to `google_client_info()`, which is expecting a string. `google.api_core.client_info.ClientInfo` seems to be what `google_client_info()` returns instead?